### PR TITLE
lara_state: add jumps when looking and look up/down while hanging

### LINKED ISSUE #848: Further look enhancements to allow standing jumps while looking and looking up/down while hanging
Lara cannot look up/down while hanging like in later classic TRs. Also, there's an old TRF request to allow forward/backward jumps while looking like in later classic TRs: https://www.tombraiderforums.com/showthread.php?p=8330600#post8330600

```
Would it be possible to amend the looking-around-while-running
functionality to behave similarly to later games, where it is possible
to make jumps from the standing position, while keeping the look
button held down?

As it is, if Lara's standing still and I have the look button held down,
Lara refuses to make any sort of jump, directional or otherwise.
This has always worked in the other classics.
```

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #848.

#### Checklist

- [X] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
Added forward/backward jumps while looking and looking up/down while hanging if enhanced look is enabled. Referenced code from [tomb3 ](https://github.com/Trxyebeep/tomb3) to make this change. Thanks to @Trxyebeep !

### CODE REVIEW

**Comment by rr-** [comment]
You can jump while hanging? What does it look like?

File: src/game/lara/lara_state.c [hunk 1]
```diff
@@ -137,12 +137,16 @@ void Lara_State_Stop(ITEM_INFO *item, COLL_INFO *coll)
...
-        return;
+        if (!g_Config.enable_enhanced_look) {
+            Lara_LookLeftRight();
+            return;
+        }
     }
-    if (g_Camera.type == CAM_LOOK) {
-        g_Camera.type = CAM_CHASE;
+    if (!g_Config.enable_enhanced_look) {
+        if (g_Camera.type == CAM_LOOK) {
```
Comment (lahm86) [inline src/game/lara/lara_state.c]: Is it worth merging the conditions?

File: src/game/lara/lara_state.c [hunk 2]
```diff
@@ -305,6 +309,12 @@ void Lara_State_FastFall(ITEM_INFO *item, COLL_INFO *coll)
 
 void Lara_State_Hang(ITEM_INFO *item, COLL_INFO *coll)
 {
+    if (g_Config.enable_enhanced_look) {
+        if (g_Input.look) {
```
Comment (lahm86) [inline src/game/lara/lara_state.c]: And similarly here

**Review by lahm86 — COMMENTED** [review]
Looks great, just a minor comment

**Comment by walkawayy** [comment]
> You can jump while hanging? What does it look like?

Lol idk if you're joking but see this video. It's two separate things.

https://github.com/LostArtefacts/Tomb1Main/assets/81546780/5dc22865-3bf5-4f9f-8259-c9c97ca4f9be

File: src/game/lara/lara_state.c (hunk 2 above)
Comment (walkawayy) [reply src/game/lara/lara_state.c]: Yes to both.

**Review by lahm86 — APPROVED** [review]

**Review by rr- — APPROVED** [review]

**Pull request merged**